test/lib/lbt/analyzer/JSModuleAnalyzer.js

```diff
@@ -684,3 +684,78 @@ jQuery.sap.registerPreloadedModules({
 	t.deepEqual(info.subModules, ["foo/bar.js"],
 		"submodule from jQuery.sap.registerPreloadedModules");
 });
+
+test("Module that contains jQuery.sap.declare should be derived as subModule", (t) => {
+	const content = `
+sap.ui.define([], function() {
+	jQuery.sap.declare("foo.bar");
+});
+`;
+	const info = analyzeString(content, "modules/module-with-jquery-sap-declare.js");
+	t.is(info.name, "modules/module-with-jquery-sap-declare.js");
+	t.is(info.rawModule, false);
+	t.is(info.format, "ui5-declare", "Format should be declare once a jQuery.sap.declare statement is found");
```

Comment:
From the content I would say that the format should be ui5-define, not ui5-declare. 

I understand that this was not changed by this PR (and rather is a sleeping bug in the analyzer). But seeing now a test that declares this the "expected" behavior, feels wrong.

Can we at least comment that this needs to be fixed? Or is there a 'todo' variant for 'test' (similar to QUnit.todo)?

Reply:
Done